### PR TITLE
Implemented creation of HA master for GCE on debian.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Implemented creation of HA master for GCE on debain.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31437)

### CODE REVIEW

**Comment by jszczepkowski** [comment]
This is follow-up of #29995.

CC @fgrzadkowski

File: cluster/gce/debian/master-helper.sh [hunk 1]
```diff
@@ -40,8 +40,31 @@ function create-master-instance {
...
+  kube_env="$(echo -e "${kube_env}\nINITIAL_ETCD_CLUSTER: '${existing_master_replicas},${REPLICA_NAME}'")"
+  echo "${kube_env}" > ${KUBE_TEMP}/master-kube-env.yaml
+  get-metadata "${existing_master_zone}" "${existing_master_name}" cluster-name > ${KUBE_TEMP}/cluster-name.txt
+  get-metadata "${existing_master_zone}" "${existing_master_name}" startup-script > ${KUBE_TEMP}/configure-vm.sh
+
+  create-master-instance-internal "${REPLICA_NAME}"
+}
+
+function create-master-instance-internal() {
+  gcloud compute instances create "${1}" \
```
Comment (wojtek-t) [inline cluster/gce/debian/master-helper.sh]: Can you please create local variables for ${1} and ${2} (similarly to what you did in GCI) - this is much more readable then.

File: cluster/gce/debian/master-helper.sh [hunk 2]
```diff
@@ -53,7 +76,18 @@ function create-master-instance {
     --can-ip-forward \
     --metadata-from-file \
       "startup-script=${KUBE_TEMP}/configure-vm.sh,kube-env=${KUBE_TEMP}/master-kube-env.yaml,cluster-name=${KUBE_TEMP}/cluster-name.txt" \
-    --disk "name=${MASTER_NAME}-pd,device-name=master-pd,mode=rw,boot=no,auto-delete=no" \
+    --disk "name=${1}-pd,device-name=master-pd,mode=rw,boot=no,auto-delete=no" \
     --boot-disk-size "${MASTER_ROOT_DISK_SIZE:-10}" \
-    ${preemptible_master}
+    ${3:-}
```
Comment (wojtek-t) [inline cluster/gce/debian/master-helper.sh]: And for ${3}.

File: cluster/gce/debian/master-helper.sh [hunk 3]
```diff
@@ -40,8 +40,31 @@ function create-master-instance {
...
+
+function replicate-master-instance() {
+  local existing_master_zone="${1}"
+  local existing_master_name="${2}"
+  local existing_master_replicas="${3}"
+
+  local kube_env="$(get-metadata "${existing_master_zone}" "${existing_master_name}" kube-env)"
+  # Substitute KUBERNETES_MASTER_NAME to the replica's name.
+  kube_env="$(echo "${kube_env}" | grep -v "KUBERNETES_MASTER_NAME")"
+  kube_env="$(echo -e "${kube_env}\nKUBERNETES_MASTER_NAME: '${REPLICA_NAME}'")"
```
Comment (wojtek-t) [inline cluster/gce/debian/master-helper.sh]: Just ensuring - it is excpect that you are replacing this here and you're not doing it in GCI, right?

File: cluster/gce/debian/master-helper.sh (hunk 1 above)
Comment (jszczepkowski) [inline cluster/gce/debian/master-helper.sh]: done

File: cluster/gce/debian/master-helper.sh (hunk 2 above)
Comment (jszczepkowski) [inline cluster/gce/debian/master-helper.sh]: done

File: cluster/gce/debian/master-helper.sh (hunk 3 above)
Comment (jszczepkowski) [inline cluster/gce/debian/master-helper.sh]: Thanks for noticing this, it is not required anymore. Removed.

**Comment by jszczepkowski** [comment]
@wojtek-t Comments applied, PTAL

**Comment by jszczepkowski** [comment]
issue: #31884
@k8s-bot test this

**Comment by wojtek-t** [comment]
LGTM

**Comment by jszczepkowski** [comment]
Part of https://github.com/kubernetes/features/issues/48

**Pull request merged**